### PR TITLE
Fix DB queries for 9.x

### PR DESCRIPTION
**Pull request by sebastian-meyer**
This pull request
- migrates the remaining usages `$GLOBALS['TYPO3_DB']` to Doctrine DBAL
- removes the then obsolete method `Helper::whereClause()`
- refactors the new method `Helper::whereExpression()` according to current TYPO3 standards

@lutzhelm Could you please review the changes to `Classes/Common/IiifManifest.php`?
@chrizzor Could you please review the changes to `Classes/Plugin/Basket.php`?

Merging this will finally fix #498 and allow us to release version 3.1.0 with full support for TYPO3 9.x.

### CODE REVIEW

File: Classes/Common/Helper.php [hunk 1]
```diff
@@ -893,31 +855,23 @@ public static function whereClause($table, $showHidden = false)
...
     public static function whereExpression($table, $showHidden = false)
     {
-        $expressionBuilder = GeneralUtility::makeInstance(ConnectionPool::class)
-            ->getQueryBuilderForTable($table)
-            ->expr();
-
         if (\TYPO3_MODE === 'FE') {
             // Should we ignore the record's hidden flag?
-            $ignoreHide = -1;
+            $ignoreHide = 0;
```
Comment (sebastian-meyer) [inline Classes/Common/Helper.php:860]: Setting `$ignoreHide` to `-1` will nowadays lead to TYPO3 using the configured page default. Therefore we have to explicitly set the value to `0` or `1`.
See method `PageRepository->enableFields()`.

File: Classes/Common/Helper.php [hunk 2]
```diff
@@ -893,31 +855,23 @@ public static function whereClause($table, $showHidden = false)
...
             if ($showHidden) {
                 $ignoreHide = 1;
             }
-
-            // $GLOBALS['TSFE']->sys_page is not always available in frontend.
-            if (is_object($GLOBALS['TSFE']->sys_page)) {
-                $expression = $GLOBALS['TSFE']->sys_page->enableFields($table, $ignoreHide);
-            } else {
-                $pageRepository = GeneralUtility::makeInstance(\TYPO3\CMS\Frontend\Page\PageRepository::class);
-                $GLOBALS['TSFE']->includeTCA();
```
Comment (sebastian-meyer) [inline Classes/Common/Helper.php:912]: There are no more edge cases in which the TCA isn't loaded in FE context. So we don't need this anymore.

File: Classes/Common/IiifManifest.php [hunk 3]
```diff
@@ -126,24 +127,35 @@ protected function establishRecordId($pid)
...
+                ->removeByType(HiddenRestriction::class);
+            $result = $queryBuilder
+                ->select('tx_dlf_metadataformat.xpath AS querypath')
+                ->from('tx_dlf_metadata')
+                ->from('tx_dlf_metadataformat')
+                ->from('tx_dlf_formats')
+                ->where(
+                    $queryBuilder->expr()->eq('tx_dlf_metadata.pid', intval($pid)),
+                    $queryBuilder->expr()->eq('tx_dlf_metadataformat.pid', intval($pid)),
+                    $queryBuilder->expr()->orX(
```
Comment (sebastian-meyer) [inline Classes/Common/IiifManifest.php:144]: Can't use a proper `join()` (or `innerJoin()` as in `MetsDocument`) because of this `OR` statement...

**Review by christopher-timm — APPROVED** [review]
Looks good to me.

**Review by albig — APPROVED** [review]
Works for me. Haven't tested IIIF and basket-feature.

**Comment by sebastian-meyer** [comment]
Thanks for your reviews!

**Pull request merged**